From 3b23f836c2ec0541aea62823eeed7f2bbe88dc43 Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Tue, 25 Apr 2023 17:15:17 -0500
Subject: [PATCH 420/508] media: platform: img: add the vxe-vxd driver into the
 build infra

Now we have a buildable driver on Linux 6.1, lets add the code
into both the Makefiles, and the configuration for it into the Kconfig.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
---
 drivers/media/platform/Kconfig              |  1 +
 drivers/media/platform/Makefile             |  1 +
 drivers/media/platform/img/Kconfig          | 35 +++++++++++++++++++++
 drivers/media/platform/img/Makefile         |  2 ++
 drivers/media/platform/img/vxe-vxd/Makefile |  2 +-
 5 files changed, 40 insertions(+), 1 deletion(-)
 create mode 100644 drivers/media/platform/img/Kconfig
 create mode 100644 drivers/media/platform/img/Makefile

diff --git a/drivers/media/platform/Kconfig b/drivers/media/platform/Kconfig
index a9334263fa9b..07a919ec4430 100644
--- a/drivers/media/platform/Kconfig
+++ b/drivers/media/platform/Kconfig
@@ -69,6 +69,7 @@ source "drivers/media/platform/aspeed/Kconfig"
 source "drivers/media/platform/atmel/Kconfig"
 source "drivers/media/platform/cadence/Kconfig"
 source "drivers/media/platform/chips-media/Kconfig"
+source "drivers/media/platform/img/Kconfig"
 source "drivers/media/platform/intel/Kconfig"
 source "drivers/media/platform/marvell/Kconfig"
 source "drivers/media/platform/mediatek/Kconfig"
diff --git a/drivers/media/platform/Makefile b/drivers/media/platform/Makefile
index a91f42024273..a43961ad2942 100644
--- a/drivers/media/platform/Makefile
+++ b/drivers/media/platform/Makefile
@@ -12,6 +12,7 @@ obj-y += aspeed/
 obj-y += atmel/
 obj-y += cadence/
 obj-y += chips-media/
+obj-y += img/
 obj-y += intel/
 obj-y += marvell/
 obj-y += mediatek/
diff --git a/drivers/media/platform/img/Kconfig b/drivers/media/platform/img/Kconfig
new file mode 100644
index 000000000000..49dd0b911d9b
--- /dev/null
+++ b/drivers/media/platform/img/Kconfig
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: GPL-2.0-only
+
+comment "IMG media platform drivers"
+
+config VIDEO_IMG_VXD_DEC
+  tristate "IMG VXD DEC (Video Decoder) driver"
+	depends on V4L_MEM2MEM_DRIVERS
+  depends on VIDEO_DEV
+  depends on ARCH_K3 || COMPILE_TEST
+  depends on V4L_PLATFORM_DRIVERS
+  select VIDEOBUF2_CORE
+  select VIDEOBUF2_DMA_CONTIG
+  select VIDEOBUF2_DMA_SG
+  select V4L2_MEM2MEM_DEV
+  help
+    This is an IMG VXD DEC V4L2 driver that adds support for the
+    Imagination D5520 (Video Decoder) hardware as used on the Texas
+    Instruments Jacinto and Sitara devices.
+    The module name when built is vxd-dec.
+
+config VIDEO_IMG_VXE_ENC
+  tristate "IMG VXE ENC (Video Encoder) driver"
+	depends on V4L_MEM2MEM_DRIVERS
+  depends on VIDEO_DEV
+  depends on ARCH_K3 || COMPILE_TEST
+  depends on V4L_PLATFORM_DRIVERS
+  select VIDEOBUF2_CORE
+  select VIDEOBUF2_DMA_CONTIG
+  select VIDEOBUF2_DMA_SG
+  select V4L2_MEM2MEM_DEV
+  help
+    This is an IMG VXE ENC V4L2 driver that adds support for the
+    Imagination VXE384 (Video Encoder) hardware as used on the Texas
+    Instruments Jacinto and Sitara devices.
+    The module name when built is vxe-enc.
\ No newline at end of file
diff --git a/drivers/media/platform/img/Makefile b/drivers/media/platform/img/Makefile
new file mode 100644
index 000000000000..57a8d7ba664d
--- /dev/null
+++ b/drivers/media/platform/img/Makefile
@@ -0,0 +1,2 @@
+# SPDX-License-Identifier: GPL-2.0-only
+obj-y += vxe-vxd/

